### PR TITLE
[DUX-1349] Start for one module

### LINKED ISSUE #69: [DUX-1349] Start for one module
`ghcid` supports being started like `ghcid scratch.hs`, which will reload a single `.hs` file.

<sub>From [SyncLinear.com](https://synclinear.com) | [DUX-1349](https://linear.app/mercury/issue/DUX-1349/start-for-one-module)</sub>

### PR DESCRIPTION
**Pull request by evanrelf**
Closes #69 (nice).

### CODE REVIEW

File: src/ghci/mod.rs [hunk 1]
```diff
@@ -127,13 +127,15 @@ impl GhciOpts {
...
-        let command = opts
-            .command
-            .clone()
-            .unwrap_or_else(|| ClonableCommand::new("cabal").arg("repl"));
+        let command = match (&opts.file, &opts.command) {
+            (Some(file), None) => ClonableCommand::new("ghci").arg(file.relative()),
+            (None, Some(command)) => command.clone(),
+            (None, None) => ClonableCommand::new("cabal").arg("repl"),
+            (Some(_), Some(_)) => unreachable!(),
+        };
```
Comment (evanrelf) [inline src/ghci/mod.rs:135]: Let me know if there's a nicer way to write this.

File: src/cli.rs [hunk 2]
```diff
@@ -54,6 +54,12 @@ use crate::normal_path::NormalPath;
     override_usage = "ghciwatch [--command SHELL_COMMAND] [--watch PATH] [OPTIONS ...]"
 )]
 pub struct Opts {
+    /// A Haskell source file to load into a `ghci` REPL.
+    ///
+    /// Shortcut for `--command 'ghci PATH'`. Conflicts with `--command`.
+    #[arg(value_name = "FILE", conflicts_with = "command")]
+    pub file: Option<NormalPath>,
```
Comment (9999years) [inline src/cli.rs:61]: Could we move this after the `command` field here so it renders after `--command` in the `--help` message?

File: src/ghci/mod.rs (hunk 1 above)
Comment (9999years) [reply src/ghci/mod.rs:135]: Maybe like `opts.command.clone().or(opts.file.map(|file| ClonableCommand::new("ghci").arg(file.relative()))).unwrap_or_else(|| ClonableCommand::new("cabal").arg("repl"))`?

Comment (9999years) [reply src/ghci/mod.rs:135]: (That will probably look better once it's formatted.)

File: src/watcher.rs [hunk 3]
```diff
@@ -38,8 +38,16 @@ impl WatcherOpts {
     /// This extracts the bits of an [`Opts`] struct relevant to the [`run_watcher`] session
     /// without cloning or taking ownership of the entire thing.
     pub fn from_cli(opts: &Opts) -> Self {
+        let watch = if let Some(file) = &opts.file {
+            let mut paths = opts.watch.paths.clone();
+            paths.push(file.clone());
+            paths
+        } else {
+            opts.watch.paths.clone()
+        };
```
Comment (9999years) [inline src/watcher.rs:47]: Let's put this in `Opts::init`:

https://github.com/MercuryTechnologies/ghciwatch/blob/2c4b72ca870559134a234831c6c59563a34631d9/src/cli.rs#L235-L237

**Review by 9999years — CHANGES_REQUESTED** [review]
Thanks for implementing this! A couple small changes and then it'll be ready for merge.

File: src/ghci/mod.rs (hunk 1 above)
Comment (evanrelf) [reply src/ghci/mod.rs:135]: Hmm, IMO that's less readable. I'll just keep it as is.

D75ZK0Y8

**Review by 9999years — APPROVED** [review]

**Pull request merged**